src/cron/SignatureRequestProcessing.ts

```diff
@@ -0,0 +1,31 @@
+import cron from "node-cron";
+
+import SignatureRequestProcessor from "../services/SignatureRequestProcessor.js";
+
+export default class SignatureRequestProcessorCron {
+  private static instance: SignatureRequestProcessorCron;
+  private processor: SignatureRequestProcessor;
+
+  private constructor() {
+    this.processor = SignatureRequestProcessor.getInstance();
+    this.setupCronJob();
+  }
+
+  private setupCronJob() {
+    // Run every 30 seconds
```

Comment:
Could it be that the previous cycle is still running? What would happen in that case? As a safeguard, could we apply an 'isRunning' flag?

Reply:
The queue handles this case. It [doesn't add commands that are already queued](https://github.com/hypercerts-org/hypercerts-api/blob/87817a44bb555389af600b4729aaa43f4d0a812e/src/services/SafeApiQueue.ts#L30-L35) and [allows only one loop](https://github.com/hypercerts-org/hypercerts-api/blob/87817a44bb555389af600b4729aaa43f4d0a812e/src/services/SafeApiQueue.ts#L42).